app/src/store/search.ts

```diff
@@ -17,7 +19,27 @@ export const searchSlice = createSlice({
     setSearchQuery: (state, { payload }: PayloadAction<string>) => {
       state.searchQuery = payload;
     },
+    setCuisineType: (state, { payload }: PayloadAction<string>) => {
+      state.cuisineType = [...payload];
```

Comment:
This confused me so much at first. It doesn't look intuitive and looks prone to mistakes.

I don't think we need `cuisineType` to be nullable. Can we change it to the following to make it easier to understand?

```
  initialState: {
    cuisineType: [] as string[],
  },
  ...
    setCuisineType: (state, { payload }: PayloadAction<string>) => {
      state.cuisineType.push(payload);
    },
```

Reply:
![image](https://user-images.githubusercontent.com/42430035/163108228-07acc6ec-77fb-4cc7-aa60-090514aa30a2.png)
If I change the **setCuisineType** to what you said, then the redux state would duplicate itself and keep adding more inner array entries. I wanted to have just one array, and add / delete the cuisine types without turning the state into a 2d array.